### PR TITLE
Generate JSON File With Linux Test AMI X Tag

### PR DESCRIPTION
**Pull request by sethAmazon**
# Description of the issue
We need a way to test integration tests independently. 

# Description of changes
Use tags for tests. Then create a generator of tags to test. 

# Tests
My fork

### CODE REVIEW

File: .github/workflows/integrationTest.yml [hunk 1]
```diff
@@ -86,6 +86,34 @@ jobs:
...
+    outputs:
+      ec2_linux_matrix: ${{ steps.set-matrix.outputs.ec2_linux_matrix }}
+      ec2_windows_matrix: ${{ steps.set-matrix.outputs.ec2_windows_matrix }}
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Go 1.x
+        uses: actions/setup-go@v2
+        with:
+          go-version: ~1.15.15
```
Comment (khanhntd) [inline .github/workflows/integrationTest.yml:101]: We might want setting this as environment variable for easier change in go version for @ZhenyuTan-amz's upcoming PR

File: .github/workflows/integrationTest.yml [hunk 2]
```diff
@@ -86,6 +86,34 @@ jobs:
...
+
+      - name: Generate matrix
+        id: set-matrix
+        run: |
+          go run --tags=generator integration/generator/test_case_generator.go
+          export ec2_linux_matrix=$(echo $(cat integration/generator/resources/linux_complete_test_matrix.json))
```
Comment (khanhntd) [inline .github/workflows/integrationTest.yml:107]: Not sure why we need to export these variables though.

**Review by khanhntd — APPROVED** [review]
Besides the previous comments, everything looks good to me.

File: .github/workflows/integrationTest.yml (hunk 1 above)
Comment (sethAmazon) [reply .github/workflows/integrationTest.yml:102]: We will just do a replace all with that pr.

**Review by khanhntd — APPROVED** [review]
LGTM.

**Pull request merged**